File: BankingInformationSystem.java
import java.util.*;

class User {
    String username;
    String password;
    double balance;

    User(String username, String password) {
        this.username = username;
        this.password = password;
        this.balance = 0.0;
    }
}

public class BankingInformationSystem {
    static Scanner scanner = new Scanner(System.in);
    static HashMap<String, User> users = new HashMap<>();

    public static void main(String[] args) {
        System.out.println("Welcome to the Banking Information System");

        while (true) {
            System.out.println("\n1. Register");
            System.out.println("2. Login");
            System.out.println("3. Exit");
            System.out.print("Choose an option: ");
            int choice = scanner.nextInt();
            scanner.nextLine(); // Consume newline

            switch (choice) {
                case 1:
                    register();
                    break;
                case 2:
                    login();
                    break;
                case 3:
                    System.out.println("Thank you for using the system!");
                    return;
                default:
                    System.out.println("Invalid option. Try again.");
            }
        }
    }

    static void register() {
        System.out.print("Enter a username: ");
        String username = scanner.nextLine();
        if (users.containsKey(username)) {
            System.out.println("Username already exists.");
            return;
        }

        System.out.print("Enter a password: ");
        String password = scanner.nextLine();
        users.put(username, new User(username, password));
        System.out.println("User registered successfully.");
    }

    static void login() {
        System.out.print("Enter username: ");
        String username = scanner.nextLine();
        System.out.print("Enter password: ");
        String password = scanner.nextLine();

        User user = users.get(username);

        if (user != null && user.password.equals(password)) {
            System.out.println("Login successful.");
            userMenu(user);
        } else {
            System.out.println("Invalid credentials.");
        }
    }

    static void userMenu(User user) {
        while (true) {
            System.out.println("\n--- Account Menu ---");
            System.out.println("1. View Balance");
            System.out.println("2. Deposit Money");
            System.out.println("3. Withdraw Money");
            System.out.println("4. Logout");
            System.out.print("Choose an option: ");
            int option = scanner.nextInt();
            scanner.nextLine(); // Consume newline

            switch (option) {
                case 1:
                    System.out.println("Current Balance: ₹" + user.balance);
                    break;
                case 2:
                    System.out.print("Enter amount to deposit: ₹");
                    double deposit = scanner.nextDouble();
                    user.balance += deposit;
                    System.out.println("Deposited ₹" + deposit + ". New balance: ₹" + user.balance);
                    break;
                case 3:
                    System.out.print("Enter amount to withdraw: ₹");
                    double withdraw = scanner.nextDouble();
                    if (withdraw > user.balance) {
                        System.out.println("Insufficient balance.");
                    } else {
                        user.balance -= withdraw;
                        System.out.println("Withdrew ₹" + withdraw + ". Remaining balance: ₹" + user.balance);
                    }
                    break;
                case 4:
                    System.out.println("Logged out successfully.");
                    return;
                default:
                    System.out.println("Invalid option.");
            }
        }
    }
}

File: BankingInformationSystem.py
import os
import json

class User:
    def __init__(self, username, password, balance=0.0):
        self.username = username
        self.password = password
        self.balance = balance

    def to_dict(self):
        return {
            'username': self.username,
            'password': self.password,
            'balance': self.balance
        }

    @staticmethod
    def from_dict(data):
        return User(data['username'], data['password'], data['balance'])

class BankingInformationSystem:
    def __init__(self, data_file='userdata.txt'):
        self.users = {}
        self.data_file = data_file
        self.load_users()

    def load_users(self):
        try:
            if os.path.exists(self.data_file):
                with open(self.data_file, 'r') as f:
                    data = json.load(f)
                    for username, user_data in data.items():
                        self.users[username] = User.from_dict(user_data)
        except Exception as e:
            print("Error loading users:", e)

    def save_users(self):
        try:
            with open(self.data_file, 'w') as f:
                json.dump({u: user.to_dict() for u, user in self.users.items()}, f)
        except Exception as e:
            print("Error saving users:", e)

    def register(self):
        username = input("Enter a username: ")
        if username in self.users:
            print("Username already exists.")
            return
        password = input("Enter a password: ")
        self.users[username] = User(username, password)
        self.save_users()
        print("User registered successfully.")

    def login(self):
        username = input("Enter username: ")
        password = input("Enter password: ")
        user = self.users.get(username)
        if user and user.password == password:
            print("Login successful.")
            self.user_menu(user)
        else:
            print("Invalid credentials.")

    def user_menu(self, user):
        while True:
            print("\n--- Account Menu ---")
            print("1. View Balance")
            print("2. Deposit Money")
            print("3. Withdraw Money")
            print("4. Logout")
            option = input("Choose an option: ")
            if option == '1':
                print(f"Current Balance: ₹{user.balance:.2f}")
            elif option == '2':
                try:
                    amount = float(input("Enter amount to deposit: ₹"))
                    user.balance += amount
                    self.save_users()
                    print(f"Deposited ₹{amount}. New balance: ₹{user.balance:.2f}")
                except ValueError:
                    print("Invalid amount.")
            elif option == '3':
                try:
                    amount = float(input("Enter amount to withdraw: ₹"))
                    if amount > user.balance:
                        print("Insufficient balance.")
                    else:
                        user.balance -= amount
                        self.save_users()
                        print(f"Withdrew ₹{amount}. Remaining balance: ₹{user.balance:.2f}")
                except ValueError:
                    print("Invalid amount.")
            elif option == '4':
                print("Logged out successfully.")
                break
            else:
                print("Invalid option.")

    def start(self):
        print("Welcome to the Banking Information System")
        while True:
            print("\n1. Register")
            print("2. Login")
            print("3. Exit")
            choice = input("Choose an option: ")
            if choice == '1':
                self.register()
            elif choice == '2':
                self.login()
            elif choice == '3':
                print("Thank you for using the system!")
                self.save_users()
                break
            else:
                print("Invalid option.")

if __name__ == "__main__":
    BankingInformationSystem().start()
